src/test/scala/mesosphere/marathon/api/akkahttp/v2/GroupsControllerTest.scala

```diff
@@ -62,6 +80,116 @@ class GroupsControllerTest extends UnitTest with ScalatestRouteTest with Inside
+      val f = new Fixture(infoService = infoService)
+
+      Get(Uri./.withPath(Path("/versions/2017-10-30T16:08:53.852Z"))) ~> f.groupsController.route ~> check {
+        responseAs[String] should include (""""id" : "/"""")
+      }
+    }
+
+    "reject for app and version that does not exist" in {
+      val infoService = mock[GroupInfoService]
+      infoService.selectGroupVersion(any, any, any, any) returns Future.successful(None)
+      val f = new Fixture(infoService = infoService)
+
+      Get(Uri./.withPath(Path("/groupname/versions/2017-10-30T16:08:53.852Z"))) ~> f.groupsController.route ~> check {
+        rejection should be (EntityNotFound.noGroup("groupname".toRootPath, Some(Timestamp("2017-10-30T16:08:53.852Z"))))
+      }
+    }
+  }
+
+  "Create a group" should {
+    "fail with rejection for group that already exists" in {
+      val groupManager = mock[GroupManager]
+      groupManager.rootGroup() returns createRootGroup()
+      val f = new Fixture(groupManager = groupManager)
+      val entity = HttpEntity("{}").withContentType(ContentTypes.`application/json`)
+
+      Post(Uri./.withPath(Path("/")), entity) ~> f.groupsController.route ~> check {
+        rejection should be (Rejections.ConflictingChange(Message(s"Group / is already created. Use PUT to change this group.")))
+      }
+    }
+
+    "fail with rejection for group name that is already an app name" in {
+      val groupManager = mock[GroupManager]
+      val rootGroup = createRootGroup(apps = Map(
+        "/appname".toRootPath -> AppDefinition("/appname".toRootPath, cmd = Some("cmd"), networks = Seq(ContainerNetwork("foo")))
+      ))
+      groupManager.rootGroup() returns rootGroup
+      val f = new Fixture(groupManager = groupManager)
+      val entity = HttpEntity("{}").withContentType(ContentTypes.`application/json`)
+
+      Post(Uri./.withPath(Path("/appname")), entity) ~> f.groupsController.route ~> check {
+        rejection should be (Rejections.ConflictingChange(Message(s"An app with the path /appname already exists.")))
+      }
+    }
+
+    "create a group" in {
+      val groupApiService = mock[GroupApiService]
+      groupApiService.updateGroup(any, any, any, any)(any) returns Future.successful(createRootGroup())
+      val groupManager = mock[GroupManager]
+      groupManager.rootGroup() returns createRootGroup()
+      groupManager.updateRoot(eq(PathId.empty), any, any, eq(false), any) returns Future.successful(DeploymentPlan.empty.copy(id = "plan", version = Timestamp.zero))
+      val f = new Fixture(groupManager = groupManager, groupApiService = groupApiService)
+      val entity = HttpEntity("{}").withContentType(ContentTypes.`application/json`)
+
+      Post(Uri./.withPath(Path("/newgroup")), entity) ~> f.groupsController.route ~> check {
+        responseAs[String] should be ("""{
```

Comment:
I know I've been doing it this way. However, I'm wondering if we should not all use the json parsing with some nice ScalaTest matchers. What do you think?

Reply:
Sounds good. Have you anything in mind or are you working on those or should I try to come up with sth?

Btw. I actually find it OK to compare response in full as string *once* for every type of response...

Reply:
I've finally put the proposal into review https://github.com/mesosphere/marathon/pull/5741.